src/Microsoft.ML.OnnxTransform/OnnxCatalog.cs

```diff
@@ -61,5 +62,19 @@ public static class OnnxCatalog
             bool fallbackToCpu = false)
         => new OnnxScoringEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnNames, inputColumnNames, modelFile, gpuDeviceId, fallbackToCpu);
 
+        /// <summary>
+        /// Applies a pre-trained DNN model to featurize an image.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="outputColumnName">The name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
+        /// <param name="modelFactory">An extension method on the <see cref="DnnImageModelSelector"/> that creates a chain of two
+        /// <see cref="OnnxScoringEstimator"/> (one for preprocessing and one with a pretrained image DNN) with specific models
+        /// included in a package together with that extension method.</param>
+        /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
+        public static DnnImageFeaturizerEstimator DnnFeaturizeImage(this TransformsCatalog catalog,
```

Comment:
>DnnFeaturizeImage [](start = 50, length = 17)

any examples for this in our samples ?

Reply:
We don't have any unfortunately. We should add it, but don't think of doing this as part of this PR. Will make sure it's in the documentation issue though.

---
In reply to: [256071655](https://github.com/dotnet/machinelearning/pull/2510#discussion_r256071655) [](ancestors = 256071655)